### ISSUE TITLE
How to generate virtual labels.

### ISSUE THREAD

**Issue by Kouuh** [kibok90/imix#4]
In SimCLR 3.2 only positive sample pairs are used, so I think the virtual label should be 1 when they belong to two different views of the same sentence, but 0 when they don't. But I still can't understand how the virtual label in (x_i, v_i) is generated.

**Comment by Kouuh** [comment]
I looked at the code that generates v_j.
`
 feature_q, labels_aux, lam = mixup(feature_q, alpha)
`
`
 def mixup(input, alpha, share_lam=False):
`
`
     ....
`
`
     randind = torch.randperm(input.shape[0], device=input.device)
`
`
    ....
`
`
    return output, randind, lam
`
`
 loss = (lam * criterion(logits, labels) + (1. - lam) * criterion(logits, labels_aux)).mean()
`
`
So can I assume that v_j is generated by randomly distorting a sequence.

**Comment by kibok90** [comment]
I assume that your question is not limited to SimCLR but about how to generate virtual labels for all methods?
As you said, we choose a random index to mix for each label; so `(i,j)` is chosen as like `zip(range(N), randperm(range(N)))`
This choice is also aligned with the original mixup implementation:
https://github.com/facebookresearch/mixup-cifar10/blob/main/train.py#L127-L132
